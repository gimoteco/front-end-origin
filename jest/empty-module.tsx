import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
