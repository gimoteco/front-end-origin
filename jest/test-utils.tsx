import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
