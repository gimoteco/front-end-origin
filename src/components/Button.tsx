import styled from 'styled-components';

export default styled.button`
  border-radius: 9999px;
  background-color: ${p => p.theme.colors.primary};
  color: ${p => p.theme.colors.white};
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5rem;
  font-family: ${p => p.theme.fonts.body};
  width: 100%;
  border: none;
  padding: 1rem;
  max-width: 400px;
`;
