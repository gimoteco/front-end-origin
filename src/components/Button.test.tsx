import React from 'react';
import { render } from '../../jest/test-utils';
import Button from './Button';

describe('Button', () => {
  it('should render according snapshot', () => {
    const { container } = render(<Button>Hello World</Button>);

    expect(container).toMatchSnapshot();
  });
});
