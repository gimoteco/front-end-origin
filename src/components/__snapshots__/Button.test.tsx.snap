// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render according snapshot 1`] = `
<div>
  <button
    class="sc-bdVaJa kquZro"
  >
    Hello World
  </button>
</div>
`;
